Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -805,6 +794,115 @@ private string GetContentRangeHeader(long chunkStart, long chunkSize)
             }
         }
 
+        #endregion Upload Implementation
+    }
+
+    /// <summary>
+    /// Media upload which uses Google's resumable media upload protocol to upload data.
+    /// </summary>
+    /// <remarks>
+    /// See: https://developers.google.com/drive/manage-uploads#resumable for more information on the protocol.
+    /// </remarks>
+    /// <typeparam name="TRequest">
+    /// The type of the body of this request. Generally this should be the metadata related to the content to be 
+    /// uploaded. Must be serializable to/from JSON.
+    /// </typeparam>
+    public class ResumableUpload<TRequest> : ResumableUpload

```

Comment:
I'm assuming all of this is basically as before - I haven't reviewed any of it carefully.